knative-operator/pkg/controller/knativeserving/kourier/kourier.go

```diff
@@ -1,53 +1,112 @@
 package kourier
 
 import (
+	"context"
+
 	mf "github.com/jcrossley3/manifestival"
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/common"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	servingv1alpha1 "knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var (
-	log       = common.Log.WithName("kourier")
-	manifests = []string{"deploy/resources/kourier/kourier-latest.yaml", "deploy/resources/kourier/kourier-openshift.yaml"}
+	log  = common.Log.WithName("kourier")
+	path = "deploy/resources/kourier/kourier-latest.yaml"
 )
 
 // Apply applies Kourier resources.
 func Apply(instance *servingv1alpha1.KnativeServing, api client.Client) error {
+	if instance.Status.IsFullySupported() {
```

Comment:
This is somewhat dangerous as we'll never reconcile the deployments again, even if the user broke them or something. We shouldn't short-circuit our own reconcilation on our own status.